Enhance workflow management with custom capabilities

ojopaul opened the pull request:
- Enhance workflow management with custom capabilities fix #1540.
 - Introduced new workflow capabilities: `edit_publishpress_workflows`, `publish_publishpress_workflows`, `unpublish_publishpress_workflows`
  - Update REST API to use workflow-specific capabilities instead of generic `edit_posts`
  - Add X-PP-Workflow-Nonce verification to all workflow endpoints for enhanced security

ojopaul committed:
- Introduced new workflow capabilities: edit_publishpress_workflows, publish_publishpress_workflows, unpublish_publishpress_workflows
- Update REST API to use workflow-specific capabilities instead of generic edit_posts
- Add X-PP-Workflow-Nonce verification to all workflow endpoints for enhanced security

ojopaul committed:
composer build:js

ojopaul committed:
Merge branch 'development' into 1540-Enhance-workflow-management-with-custom-capabilities

andergmartins commented on src/Modules/Workflows/Rest/RestApiV1.php:
```diff
@@ -379,6 +384,38 @@ public function getWorkflowsWithManualTrigger($request)
 
     public function checkUserCanCallApi($request)
```
This unique method for checking all the routes makes unclear what is the specific required capability for each route without reading the conditional logic inside the method. We should simplify this, even we end up with more code. Lets aim for something more clear like defining the capability check in the route registration. Each route, with its own permissions check function.

```php
// In your endpoint registration
register_rest_route($namespace, '/workflows', [
    'methods' => 'POST',
    'callback' => [$this, 'createWorkflow'],
    'permission_callback' => function($request) {
        return $this->hasValidNonce($request)
            && current_user_can(self::PERMISSION_CREATE);
    }
]);

register_rest_route($namespace, '/workflows/(?P<id>\d+)', [
    'methods' => 'POST',
    'callback' => [$this, 'updateWorkflow'],
    'permission_callback' => function($request) {
        return $this->hasValidNonce($request)
            && current_user_can(self::PERMISSION_UPDATE);
    }
]);

...
```

Also, lets move the nonce validation to another private method `hasValidNonce` and call it on each permission check function.

```php
/**
     * Validate the nonce for the request
     *
     * @param WP_REST_Request $request The request object
     *
     * @return bool True if the nonce is valid, false otherwise
     */
    private function hasValidNonce($request)
    {
        $nonce = $request->get_header('X-WP-Nonce');
        if (!$nonce || !wp_verify_nonce($nonce, 'wp_rest')) {
            return false;
        }

        $workflowNonce = $request->get_header('X-PP-Workflow-Nonce');
        if (!$workflowNonce || !wp_verify_nonce($workflowNonce, 'pp_workflow_action')) {
            return false;
        }

        return true;
    }
```

ojopaul replied:
Thanks, on it...

andergmartins reviewed (changes requested):
Just need an adjustment to the rest api permission check for better code readbility and clearer permissions visualization.

ojopaul committed:
Simplify rest api permission callback

ojopaul committed:
Format code

andergmartins reviewed (changes requested):
@ojopaul the permissions check is working great, but it is missing the implementation to grant the new capabilities to the administrators. Try it on a fresh site, you will see you can't edit or create workflows. That should be done when we detect a plugin activation, and avoid to overwrite changes made by users using PublishPress Capacities (by removing capability from specific users/roles).

<img width="432" height="172" alt="image" src="https://github.com/user-attachments/assets/309867da-280b-4d57-a985-5bde65c48861" />

ojopaul committed:
Merge branch 'development' into 1540-Enhance-workflow-management-with-custom-capabilities

ojopaul committed:
- Update post type registration to use custom capabilities
- Add capability initialization for administrators
- Update workflow editor menu to use EDIT_WORKFLOWS capability instead of manage_options

ojopaul commented:
Thanks @andergmartins I've updated the PR.
